internal/utils/utils.go

```diff
@@ -0,0 +1,232 @@
+package utils
+
+import (
+	_ "embed"
+
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/paketo-community/ubi-nodejs-extension/structs"
+
+	"github.com/BurntSushi/toml"
+)
+
+var DEFAULT_USER_ID = 1002
+var DEFAULT_GROUP_ID = 1000
+
+//go:embed templates/build.Dockerfile
+var buildDockerfileTemplate string
+
+//go:embed templates/run.Dockerfile
+var runDockerfileTemplate string
+
+type StackImages struct {
+	Name              string `json:"name"`
+	IsDefaultRunImage bool   `json:"is_default_run_image,omitempty"`
+	NodeVersion       string
+}
+
+type ImagesJson struct {
+	StackImages []StackImages `json:"images"`
+}
+
+func GenerateConfigTomlContentFromImagesJson(imagesJsonPath string, stackId string) ([]byte, error) {
+	imagesJsonData, err := ParseImagesJsonFile(imagesJsonPath)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	nodejsStacks, err := GetNodejsStackImages(imagesJsonData)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	defaultNodeVersion, err := GetDefaultNodeVersion(nodejsStacks)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	configTomlContent, err := CreateConfigTomlFileContent(defaultNodeVersion, nodejsStacks, stackId)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	configTomlContentString := configTomlContent.Bytes()
+	return configTomlContentString, nil
+}
```

Comment:
Added a wrapper, and for that wrapper on the utils tests we only test that it manages to return the result and that it manages to propagate an error. The type of the error, the type of the result and the edge cases are covered from the tests of the functions that are wrapped.